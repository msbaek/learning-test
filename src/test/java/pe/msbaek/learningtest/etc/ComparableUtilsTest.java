package pe.msbaek.learningtest.etc;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.math.BigDecimal;

import static org.apache.commons.lang3.compare.ComparableUtils.is;

public class ComparableUtilsTest {
    @DisplayName("commons-lang3의 ComparableUtils를 사용한 비교 테스트")
    @Test
    void compare_using_comparable_utils() {
        BigDecimal valueA = new BigDecimal("1.0");
        BigDecimal valueB = new BigDecimal("2.0");

        boolean isEqual = is(valueA).equalTo(valueB);
        boolean isGreater = is(valueA).greaterThan(valueB);
        boolean isGreaterThanOrEqualTo = is(valueA).greaterThanOrEqualTo(valueB);
        boolean isLess = is(valueA).lessThan(valueB);
        boolean isLessThanOrEqualTo = is(valueA).lessThanOrEqualTo(valueB);

        String s = """
                valueA = %f, valueB = %f
                
                is(valueA).equalTo(valueB)              = %b
                is(valueA).greaterThan(valueB)          = %b
                is(valueA).greaterThanOrEqualTo(valueB) = %b
                is(valueA).lessThan(valueB)             = %b
                is(valueA).lessThanOrEqualTo(valueB)    = %b
                """.formatted(valueA, valueB, isEqual, isGreater, isGreaterThanOrEqualTo, isLess, isLessThanOrEqualTo);
        Approvals.verify(s);
    }
}