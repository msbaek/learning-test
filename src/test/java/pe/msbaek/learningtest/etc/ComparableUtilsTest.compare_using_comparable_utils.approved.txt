valueA = 1.000000, valueB = 2.000000

is(valueA).equalTo(valueB)              = false
is(valueA).greaterThan(valueB)          = false
is(valueA).greaterThanOrEqualTo(valueB) = false
is(valueA).lessThan(valueB)             = true
is(valueA).lessThanOrEqualTo(valueB)    = true
